packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -230,33 +244,89 @@ const service = (dependencies: BaseDependencies) => {
+	 *   If an error occurs during data retrieval, `metadata` will include an `errorMessage` property.
+	 */
 	const getSubmittedDataByCategory = async (
 		categoryId: number,
 		paginationOptions: PaginationOptions,
-		filterOptions: { entityName?: (string | undefined)[] },
+		filterOptions: { entityName?: (string | undefined)[]; view: ViewType },
 	): Promise<{
 		data: SubmittedDataResponse[];
 		metadata: { totalRecords: number; errorMessage?: string };
 	}> => {
 		const { getSubmittedDataByCategoryIdPaginated, getTotalRecordsByCategoryId } = submittedDataRepo;
 
-		const { categoryIdExists } = categoryRepository(dependencies);
+		const { getCategoryById } = categoryRepository(dependencies);
 
-		const isValidCategory = await categoryIdExists(categoryId);
+		const category = await getCategoryById(categoryId);
 
-		if (!isValidCategory) {
+		if (!category?.activeDictionary) {
 			return fetchDataErrorResponse(PAGINATION_ERROR_MESSAGES.INVALID_CATEGORY_ID);
 		}
 
-		const recordsPaginated = await getSubmittedDataByCategoryIdPaginated(categoryId, paginationOptions, {
-			entityNames: filterOptions?.entityName,
+		let recordsPaginated = await getSubmittedDataByCategoryIdPaginated(categoryId, paginationOptions, {
+			entityNames: filterQueryByEntityName(filterOptions, category.defaultCentricEntity),
 		});
 
 		if (recordsPaginated.length === 0) {
 			return fetchDataErrorResponse(PAGINATION_ERROR_MESSAGES.NO_DATA_FOUND);
 		}
 
-		const totalRecords = await getTotalRecordsByCategoryId(categoryId, { entityNames: filterOptions?.entityName });
+		if (filterOptions.view === VIEW_TYPE.Values.compound) {
+			// get dictionary hierarchy structure
+			const hierarchyStructureDesc = generateHierarchy(category.activeDictionary.dictionary, ORDER_TYPE.Values.desc);
+			const hierarchyStructureAsc = generateHierarchy(category.activeDictionary.dictionary, ORDER_TYPE.Values.asc);
+
+			recordsPaginated = await Promise.all(
+				recordsPaginated.map(async (record) => {
+					const childNodes = await traverseChildNodes({
+						data: record.data,
+						entityName: record.entityName,
+						organization: record.organization,
+						schemaCentric: getSchemaForCompound({
+							filterByEntityName: filterOptions?.entityName,
+							schemaCentricEntity: category.defaultCentricEntity,
+							recordEntityName: record.entityName,
+						}),
+						treeNode: hierarchyStructureDesc,
+					});
+
+					const parentNodes = await traverseParentNodes({
+						data: record.data,
+						entityName: record.entityName,
+						organization: record.organization,
+						schemaCentric: getSchemaForCompound({
+							filterByEntityName: filterOptions?.entityName,
+							schemaCentricEntity: category.defaultCentricEntity,
+							recordEntityName: record.entityName,
+						}),
+						treeNode: hierarchyStructureAsc,
+					});
+
+					record.data = { ...record.data, ...childNodes, ...parentNodes };
+					return record;
+				}),
+			);
+		}
```

Comment:
Added a try catch block within this map iteration to handle errors by logging the record ID and error message. This ensures that if a record fails to convert to a compound document, the plain document is returned instead

Reply:
Would we want the flat document if we fail to build the joins? I think it's important to know that the join attempt failed instead of getting data that implies that there was nothing to join with.